### DISCUSSION TITLE
How to setup pp_defs to ignore the file extension case

### DISCUSSION THREAD

**Discussion by sdsnatcher** [fortran-lang/fortls#339] [Q&A]
The "Preprocessor: Suffixes" settings help text explains that:

"By default preprocessor definitions are parsed for all Fortran source files with uppercase extensions e.g. .F90, .F, .F08, etc.. However, the default behaviour can be overridden by defining pp_defs"

Sadly, the pp_defs help text links to a page that isn't found anymore. I tried to google for how to configure this, and found examples of some other keys to configure pp_devs like "HAVE_HDF5" and "MPI_Comm", but nothing related to how to change the behaviour to also parse lowercase extensions.

So, how should be pp_defs setup for this?

**Comment by gnikit** [comment]
which link are you referring to? if referring to VS Code's Modern Fortran install the pre-release. Links to fortls have been updated there
FYI the fortls docs are located in https://fortls.fortran-lang.org/ , I suspect you are looking for the options under https://fortls.fortran-lang.org/options.html#preprocessor

**Reply by sdsnatcher** [reply]
For the not found links, I'm referring to the ones in VSCode's Modern Fortran 3.2.0 preferences pane.

For the page I found while googling, yes it's the https://fortls.fortran-lang.org/options.html#preprocessor , but this one doesn't show how to configure the pp_defs to ignore the file extension case as I need.  

So, my main question is still unanswered: how should be pp_defs setup for this?
